migrations/promodb/000_UNIFIED_SCHEMA.sh

```
@@ -1,216 +1,151 @@
-run_sql "GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO $DB_USER;" "Grant sequence privileges"
-run_sql "ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT ALL ON TABLES TO $DB_USER;" "Set default table privileges"
-run_sql "ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT ALL ON SEQUENCES TO $DB_USER;" "Set default sequence privileges"
+run_sql "
+CREATE TABLE IF NOT EXISTS schema_migration_audit (
+  id BIGSERIAL PRIMARY KEY,
+  migration_file TEXT NOT NULL UNIQUE,
+  applied_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+" 'Ensure schema_migration_audit exists'
+
+for applied in $(find "$MIGRATION_DIR" "$FIXES_DIR/PROMO_DB" -maxdepth 1 -type f -name '*.sql' | xargs -n1 basename | sort); do
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)